lib/archethic_web/live/chains/node_shared_secrets_live.ex

```diff
@@ -0,0 +1,221 @@
+          |> assign(:nb_authorized_nodes, 0)
+          |> assign(:current_page, 1)
+          |> assign(:transactions, [])
+
+        address when is_binary(address) ->
+          nb_authorized_nodes =
+            address
+            |> TransactionChain.get_last_transaction()
+            |> elem(1)
+            |> get_in([Access.key(:data), Access.key(:ownerships)])
+            |> Enum.at(0)
+            |> get_in([Access.key(:authorized_keys)])
+            |> Enum.count()
+
+          socket
+          |> assign(:tx_count, tx_count)
+          |> assign(:nb_pages, total_pages(tx_count))
+          |> assign(:nb_authorized_nodes, nb_authorized_nodes)
+          |> assign(:current_page, 1)
+          |> assign(:transactions, transactions_from_page(1, tx_count))
+      end
+
+    {:ok, socket}
+  end
+
+  @spec render(assigns :: LiveView.Socket.assigns()) :: LiveView.Rendered.t()
+  def render(assigns) do
+    View.render(ExplorerView, "node_shared_secrets_chain_index.html", assigns)
+  end
+
+  @spec handle_params(_params :: map(), _uri :: binary(), socket :: LiveView.Socket.t()) ::
+          {:noreply, LiveView.Socket.t()}
+  def handle_params(
+        _params = %{"page" => page},
+        _uri,
+        socket = %{assigns: %{nb_pages: nb_pages, tx_count: tx_count}}
+      ) do
+    case Integer.parse(page) do
+      {number, ""} when number < 1 and number > nb_pages ->
+        {:noreply, push_patch(socket, to: Routes.live_path(socket, __MODULE__, %{"page" => 1}))}
+
+      {number, ""} when number > 0 and number <= nb_pages ->
+        socket =
+          socket
+          |> assign(:current_page, number)
+          |> assign(:transactions, transactions_from_page(number, tx_count))
+
+        {:noreply, socket}
+
+      _ ->
+        {:noreply, socket}
+    end
+  end
+
+  def handle_params(_, _, socket) do
+    {:noreply, socket}
+  end
+
+  @spec handle_event(_event :: binary(), _params :: map(), socket :: LiveView.Socket.t()) ::
+          {:noreply, LiveView.Socket.t()}
+  def handle_event(_event = "next_page", _params = %{"page" => page}, socket) do
+    {:noreply, push_patch(socket, to: Routes.live_path(socket, __MODULE__, %{"page" => page}))}
+  end
+
+  def handle_event(_event = "prev_page", _params = %{"page" => page}, socket) do
+    {:noreply, push_patch(socket, to: Routes.live_path(socket, __MODULE__, %{"page" => page}))}
+  end
+
+  @spec handle_info(
+          _msg ::
+            {:new_transaction, address :: binary(), :node_shared_secrets, DateTime.t()},
+          socket :: LiveView.Socket.t()
+        ) ::
+          {:noreply, LiveView.Socket.t()}
+  def handle_info(
+        _msg = {:new_transaction, address, :node_shared_secrets, timestamp},
+        socket = %{assigns: %{current_page: current_page, transactions: txs, tx_count: tx_count}}
+      ) do
+    display_txs = Enum.count(txs)
+
+    updated_socket =
+      case current_page do
+        1 when display_txs < @display_limit ->
```

Comment:
There is an inconsistent behavior there, when we already display 10 tx, when a new one appear the page only display this one, but if we refresh the page it display the last 10 tx.
I suggest to always display the last 10 tx, so on new_transaction you can do this
```
socket
|> update(
  :transactions,
  &([display_data(address, nb_authorized_nodes, timestamp) | &1]
    |> Enum.take(@display_limit))
)
```
It will keep the 10 last transaction (in reverse) and put the oldest transaction in next pages
I think this behavior should also be applied to Reward chain and Origin chain explorer